### PR TITLE
Add recipe for pythonfmu

### PR DESCRIPTION
**Pull request by fcollonval**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
Sorry if I'm intruding, but it looks like `cmake` is picking up the wrong `python` interpreter. Can you patch the `setup.py` or the cmake configuration to specify `-DPython3_EXECUTABLE:FILEPATH="${PYTHON}"` (and similar for windows)?

**Comment by fcollonval** [comment]
@duncanmmacleod Thanks for the tip. 

Actually I put this one on hold as I worked on the [base package](https://github.com/NTNU-IHB/PythonFMU) to build against the Python limited API to be compatible to all 3.x versions with the same binary.

This is now done. And I'll proceed with publication on pypi and then on this recipe with (normally) no more need for compilation here.

**Comment by fcollonval** [comment]
@conda-forge/help-python-c could you help me with this recipe, please?

This package needs to distribute dll compiled using `Py_LIMITED_API` so that archive created with this Python package are usable on multiple Python versions. On the project CI, everything runs as expected - meaning building the dll once per OS and testing against multiple Python versions (see [CI script](https://github.com/NTNU-IHB/PythonFMU/blob/master/.github/workflows/main.yml) and [CI trace](https://github.com/NTNU-IHB/PythonFMU/actions?query=workflow%3ACI)).  
But here the `error-overlinking` option is blocking the linux test. From the documentation, it seems `build:run_exports` may offer a solution. But I don't know how to set that parameter; especially as `noarch` does not allow selector.

**Comment by fcollonval** [comment]
@conda-forge/help-python-c this is ready for review

File: recipes/pythonfmu/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    - pip
+  run:
+    - python
+
+test:
+  requires:
+    - pytest
+    - pyfmi
+  commands:
+    - pytest --pyargs pythonfmu.tests
```
Comment (marcelotrevisani) [inline recipes/pythonfmu/meta.yaml]:
```suggestion
    - pytest --pyargs pythonfmu.tests
  imports:
    - pythonfmu
```

**Comment by fcollonval** [comment]
Thanks for the review @marcelotrevisani and for the suggestion.

Could you merge this PR please?

File: recipes/pythonfmu/bld.bat [hunk 2]
```diff
@@ -0,0 +1,9 @@
+mkdir tmp-build
```
Comment (scopatz) [inline recipes/pythonfmu/bld.bat]: You need `if errorlevel 1 exit 1`  after every command here

File: recipes/pythonfmu/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  summary: A lightweight framework that enables the packaging of Python3.x code as co-simulation FMUs.
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  dev_url: https://github.com/NTNU-IHB/PythonFMU
+
+extra:
+  recipe-maintainers:
+    - fcollonval
+    - markaren
```
Comment (scopatz) [inline recipes/pythonfmu/meta.yaml]: @markaren are you OK with being a maintainer?

Comment (markaren) [reply recipes/pythonfmu/meta.yaml]: yes

**Comment by fcollonval** [comment]
@scopatz Thanks for reviewing the recipe. I apply your recommendations and thanks @markaren for your quick reply.

**Comment by scopatz** [comment]
It seems like it is still failing to build on windows

**Comment by fcollonval** [comment]
@scopatz After harder time than expected, here is a working version.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**